commands/dns-records.go

```diff
@@ -74,11 +129,142 @@ func (cmd *DNSRecords) LoadRecords() ([]map[string]interface{}, error) {
 	}
 
 	dnsdockMap, _ := js.Map()
-	records := []map[string]interface{}{}
-	for id, value := range dnsdockMap {
-		record := value.(map[string]interface{})
-		record["Id"] = id
+	records := make([]*DNSRecord, 0, 20)
```

Comment:
Interested in your thoughts behind the use of the capacity argument here. As near as I can tell this doesn't set any upper bound on the size of the array though I had to write a quick test to make sure. Is the initial capacity you picked arbitrary (both here and below where you used 10) or is there something behind that particular value?

Reply:
Completely arbitrary. Yes, when accumulating the array contents via `append()` more space will be allocated if needed.

Reply:
Given that it's a dynamic array perhaps just drop the capacity argument here and anywhere it doesn't have an obvious purpose. This is to prevent confusing anyone in the future about why it is there, how it got picked, does it actually result in enforced bounds, etc? Alternatively, comment something to the effect of what I think may be true about it:

// capacity picked arbitrarily with a size likely to prevent need to reallocate underlying arrays while appending items in most cases

I also presume that has some potential performance benefit though at the scales here I think it's unobservable.

If that comment isn't accurate I'd be interested in what the benefits are of declaring a capacity are in the case where append is going to be used. When using copy it's an obvious useful item to restrict the number of items you want from a potentially larger array.

Thanks for the pointers as I drag myself deeper into go.